File: frontend/src/mocks/mockToken.ts
function encodePart(value: unknown): string {
  return btoa(JSON.stringify(value)).replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '');
}

/** JWT sintético para modo demo (sin verificación de firma; solo UI de roles). */
export function buildMockJwt(username: string, email: string, roles: string[]): string {
  const header = encodePart({ alg: 'none', typ: 'JWT' });
  const payload = encodePart({
    sub: username,
    preferred_username: username,
    email,
    realm_access: { roles },
  });
  return `${header}.${payload}.mock-signature`;
}

File: mobile/lib/mocks/mock_token.dart
import 'dart:convert';

import 'mock_demo_profile.dart';

/// JWT sintético para modo demo (sin verificación de firma; solo UI de roles).
String buildMockJwt({
  required String username,
  required String email,
  required List<String> roles,
}) {
  String encodePart(Object value) {
    final normalized = base64Url.encode(utf8.encode(jsonEncode(value)));
    return normalized.replaceAll('=', '');
  }

  final header = encodePart({'alg': 'none', 'typ': 'JWT'});
  final payload = encodePart({
    'sub': username,
    'preferred_username': username,
    'email': email,
    'realm_access': {'roles': roles},
  });
  return '$header.$payload.mock-signature';
}

String mockJwtForProfile(MockDemoProfile profile) => buildMockJwt(
      username: profile.username,
      email: profile.email,
      roles: profile.roles,
    );
